README.md

```diff
@@ -558,6 +559,31 @@ bus
   );
 ```
 
+### Lambda
+
+Functionless supports the use of native typescript callbacks as lambda function handlers.
+
+```ts
+new Function(this, 'myFunction', async (event) => {
+  console.log(event);
+  return Object.keys(event).length;
+})
+```
+
+#### !!CAVEAT: Using Function Callbacks with `app.synth()`!!
+
+Normal use of the CDK through the CLI should work just fine without doing anything special.
+
+The problem comes when using the explicit `app.synth()` method. This is a common case if trying to test your CDK code through testing tooling like `jest`.
+
+CDK does not natively support async code through the constructs, however, Functionless is using `pulumi`'s `serializeFunction` which is an async function. 
+
+When using `.synth()` programmatically, use the provided `asyncSynth` method to wrap your app.
+
+```ts
+const cloudAssembly = asyncSynth(app, options);
```

Comment:
Maybe show a test case for clarity. Also, should this be `await asyncSynth`?

Reply:
Not sure if I want to show a full test, actually deploying the template is verbose. Will link to an example.